basalt/autograd/node.mojo

```diff
@@ -10,38 +10,35 @@ from .attributes import AttributeVector
 @value
 struct Node(CollectionElement, Stringable):
     var operator: OP
-    var output: Symbol
-    var input_1: Symbol
-    var input_2: Optional[Symbol]
-    var input_3: Optional[Symbol]
+    var inputs: List[Symbol]
+    var outputs: List[Symbol]
```

Comment:
Have we considered a different type other than List since it has so much performance overhead?

Reply:
Since this is comptime, the only alternative would be to have a `StaticTuple[Symble, MAX_...]()` with a fixed max length, I think. It also complicates knowing how many inputs and outputs you actually have in the node. We could try to see what the effect on performance is. But it would make to code worse i think.

Reply:
I agree I feel the code would be worse and I also think List will get better in the future, i think those performance problems is better to wait, and even if in the future there is a little overhead in the end, i think it can be a good compromise, at the end the time overhead it would add (today or in the future I think) is very little or nothing when training the model (kernel and memory management and movement are what affect the performance the most in reality